src/channel_request/channel_manager.rs

```diff
@@ -0,0 +1,892 @@
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::ops::Deref;
+use std::sync::{Arc, Mutex, RwLock};
+
+use bitcoin::hashes::Hash;
+use bitcoin::secp256k1::PublicKey;
+use lightning::ln::msgs::{
+	ChannelMessageHandler, ErrorAction, LightningError, OnionMessageHandler, RoutingMessageHandler,
+};
+use lightning::ln::peer_handler::{
+	APeerManager, CustomMessageHandler, PeerManager, SocketDescriptor,
+};
+use lightning::routing::gossip::NetworkGraph;
+use lightning::sign::{EntropySource, NodeSigner};
+use lightning::util::errors::APIError;
+use lightning::util::logger::{Level, Logger};
+
+use crate::channel_request::msgs::{CreateOrderRequest, Message, Order, Request};
+use crate::transport::message_handler::ProtocolMessageHandler;
+use crate::transport::msgs::{LSPSMessage, RequestId};
+use crate::utils;
+use crate::{events::Event, transport::msgs::ResponseError};
+
+use super::event;
+use super::msgs::{
+	ChannelInfo, CreateOrderResponse, GetInfoRequest, GetInfoResponse, GetOrderRequest,
+	GetOrderResponse, OnchainPayment, OptionsSupported, OrderId, OrderState, Payment, PaymentState,
+	Response,
+};
+use super::utils::check_if_valid;
+
+const SUPPORTED_SPEC_VERSION: u16 = 1;
+
+#[derive(PartialEq)]
+enum ChannelState {
+	InfoRequested,
+	OrderRequested,
+	PendingSelection,
+	PendingPayment,
+	Ready,
+}
+
+struct CRchannel {
```

Comment:
nit: Should name this `CRChannel` probably.
```suggestion
struct CRChannel {
```